### ISSUE TITLE
[Mac] Opening two videos in quick succession on the same MediaPlayer sometimes keeps playing the first one

### ISSUE THREAD

**Issue by TheloTheGreat** [RenderHeads/UnityPlugin-AVProVideo#704]
On Mac AVPro 2.0.10, but not on Windows, if you open a video A, then immediately open a video B on the same MediaPlayer, the MediaPlayer sometimes keeps playing video A rather than video B. It should always keep playing video B, which is the last one requested.

Repro: On Demo_MediaPlayer, add this to Update:

```
if (Input.GetKeyDown(KeyCode.A)) {
	_mediaPlayer.OpenMedia(MediaPathType.RelativeToProjectFolder, "Assets/StreamingAssets/AVProVideoSamples/BigBuckBunny-360p30-H264.mp4");
	_mediaPlayer.OpenMedia(MediaPathType.RelativeToProjectFolder, "Assets/StreamingAssets/AVProVideoSamples/Cones-2D-1080p60-H264.mp4");
}
```

Then launch the demo and press A. You may need to press it several times before seeing the issue.

Expected result: The Cones-2D video always plays.
Actual result: About 2 times out of 10, the BigBuckBunny video plays.

Note: This is Mac-only. On Windows, the issue never happens.

(Also, 2 times out of about 300 in my testing, the entire Unity editor crashed, but that's inconsistent so it might not be related.)

**Comment by MorrisRH** [comment]
Open is an asynchronous operation on macOS so whichever call to open completes last will be the video that gets played and if they're both able to complete then you get a mix of the two videos in the player.
A fix will be in the next release.

**Comment by MorrisRH** [comment]
Hey @TheloTheGreat 
Would be great if you could confirm this is fixed in 2.1.0, thanks!

**Comment by TheloTheGreat** [comment]
Yes, confirmed, it's fixed in 2.1.0, thanks.
